avocado/core/extension_manager.py

```diff
@@ -70,7 +70,8 @@ def __init__(self, namespace, invoke_kwds=None):
                     self.extensions.append(ext)
         self.extensions.sort(key=lambda x: x.name)
 
-    def enabled(self, extension):  # pylint: disable=W0613
+    @staticmethod
+    def enabled(extension):  # pylint: disable=W0613
```

Comment:
Isn't it an abstract method instead of a static method, just like the docstring suggests? It is reimplemented here: https://github.com/avocado-framework/avocado/blob/e32da80252bf17fcda8cacbca0bb6af229016453/avocado/core/enabled_extension_manager.py#L40.

Reply:
Hi @willianrampazzo, I decided to go with ``staticmethod`` here as I think using ``abstractmethod`` would enforce ``enabled`` to always be implemented and that would require a modification in ``SettingsDispatcher`` as well. But, I think this is definitely a valid point of discussion, what you are suggesting for sure makes sense for me too.

Reply:
I think touching this class now may not be a good idea. Maybe we disable the lint locally, then discuss what should be done, then fix it later. What do you think? This way, we won't hold your PR.

Reply:
Sounds good! I'm planning to do an follow up in #3466 sometime in the future to give an update and to bring some ideas to discussion, I guess we could bring this one up as well.